### PR TITLE
address them all

### PR DESCRIPTION
**Pull request by casperdcl**
addresses:
- #175 (retry perf tests on failure)
- #176 (lazy import notebook)
- #177 (PR for above)
- #178 (colorama only on win)

### CODE REVIEW

**Comment by lrq3000** [comment]
Ow, that's interesting, `tqdm_notebook` is not at all covered now, is it because of the pragma no cover in the adapters or because the import is correctly delayed and thus coverage can't access the module?

I agree with this PR, it can be merged. I would prefer to generalize the delayed/lazy import for all modules, but that will do for a quick fix, it can be enhanced later.

File: tqdm/_utils.py [hunk 1]
```diff
@@ -1,5 +1,12 @@
 import os
 import subprocess
+from platform import system as _curos
+CUR_OS = _curos()
+IS_WIN = CUR_OS in ['Windows', 'cli']
+IS_NIX = (not IS_WIN) and any(
+    CUR_OS.startswith(i) for i in
+    ['CYGWIN', 'MSYS', 'Linux', 'Darwin', 'SunOS', 'FreeBSD'])
```
Comment (yarikoptic) [inline tqdm/_utils.py]: if I were you I would have added `.lower()` and kept actual literals for comparison in lower case.

-- said yarikoptic and then looked at his own code which doesn't (yet) do that ;)

Comment (yarikoptic) [inline tqdm/_utils.py]: wouldn't it be safer just to `IS_NIX = not IS_WIN`? ;-)

**Comment by casperdcl** [comment]
I'm not a fan of adding pragma no cov to code which strictly speaking should be covered... Is there no way to test widgets? redirect html output etc?

File: tqdm/_utils.py (hunk 1 above)
Comment (casperdcl) [inline tqdm/_utils.py]: Wouldn't want `lower` to encourage consistency. `platform.system()` should never equal, for example, `windows`. Also we can't assume all systems which we fail to detect as windows systems are unix. I'd be happier to add "if neither nix nor win, print warning"

**Comment by lrq3000** [comment]
@casperdcl Sorry for the delayed answer. Yes it is possible but way too complicated at this stage for me. We need to simulate a fake JS server and IPython API mockup, like the ipywidgets project does. I tried to replicate what they did but it was way over my head...

**Pull request merged**